File: 0079-word-search/0079-word-search.java
class Solution {
    boolean b[][];
    public boolean exist(char[][] board, String word) {
        b=new boolean[board.length][board[0].length];
        for(int i=0;i<board.length;i++){
            for(int j=0;j<board[0].length;j++){
                if(board[i][j]==word.charAt(0)){
                   if (solve(i,j,word,board,0)){
                    return true;
                   }
                }
            }
        }
        return false;
    }
    boolean solve(int i,int j,String word,char a[][],int ind){
        if(ind>=word.length()){
            return true;
        }
        if (i < 0 || j < 0 || i >= a.length || j >= a[0].length
                || a[i][j] != word.charAt(ind)||b[i][j]) {
            return false;
        }
        b[i][j]=true;
      boolean t=  solve(i+1,j,word,a,ind+1)||solve(i-1,j,word,a,ind+1)||solve(i,j+1,word,a,ind+1)||solve(i,j-1,word,a,ind+1);
        b[i][j]=false;
        return t;


    }
}

File: 0079-word-search/0079-word-search.py
class Solution:
    def exist(self, board: List[List[str]], word: str) -> bool:
        row=len(board)
        col=len(board[0])
        a = [[0]*col for _ in range(row)]
        for i in range(row):
            for j in range(col):
                if board[i][j]==word[0]:
                    if self.solve(a,board,i,j,word,0):
                        return True
        return False
    def solve(self,a,board,i,j,word,c):
        if (c==len(word)):
            return True
        if( i<0 or j<0 or j>=len(board[0]) or i>=len(board) or a[i][j]==1 or board[i][j]!=word[c]):
            return False
        a[i][j]=1
        got =self.solve(a,board,i-1,j,word,c+1) or self.solve(a,board,i+1,j,word,c+1) or self.solve(a,board,i,j+1,word,c+1)or self.solve(a,board,i,j-1,word,c+1)
        a[i][j]=0
        return got
        